File: 0055-jump-game/0055-jump-game.go
func canJump(nums []int) bool {
    lastIndex := len(nums) - 1
    
    for i := lastIndex; i >= 0; i-- {
        if i + nums[i] >= lastIndex {
            lastIndex = i
        }
    }
    
    if lastIndex == 0 {
        return true
    }
    
    return false
}

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int lastIndex = nums.length - 1;

        for(int i = lastIndex; i >= 0; i--) {
            if(i + nums[i] >= lastIndex) {
                lastIndex = i;
            }
        }

        if(lastIndex == 0) {
            return true;
        }
        
        return false;
    }
}